### ISSUE TITLE
Small wording change for wallet creation date

### ISSUE THREAD

**Issue by Realchacal** [monero-project/monero-gui#2328]
Currently:
![image](https://user-images.githubusercontent.com/46682965/62125562-c7b3cd00-b2cd-11e9-86de-f8ff22c0922f.png)

- User thinks **"Shit!! Nobody told me to safe the date!!"**, panics.
- One could wonder if the "restore height" is `current block height` minus `wallet creation block height` or just the `wallet creation block height`.
- When the default value ("0") is a **block height** then the label should **first** mention the block height **then** the date.
- According to the label one would have to enter the block height without backticks but the date with backticks.

Proposal (solves all 4):
![image](https://user-images.githubusercontent.com/46682965/62125689-1d887500-b2ce-11e9-93a7-660baf1a64ae.png)

______________________

These changes should also be applied for the regular restore process (without hardware wallet). Currently looks like:
![image](https://user-images.githubusercontent.com/46682965/62127259-e87e2180-b2d1-11e9-82a6-103db77c105f.png)
"0" should be the default here as well.

Alternative:
![image](https://user-images.githubusercontent.com/46682965/62128606-0f8a2280-b2d5-11e9-8260-e707d1654e56.png)
IMO better. In any case, it should be the same with and without hardware wallet.

**Comment by GBKS** [comment]
I agree that this need so be simplified. "Restore height" made no sense to me initially, I didn't even realize it was about block numbers, and didn't know what the relevance of this option even was. "Wallet creation date" would work. The GUI can always start scanning the blockchain 1-2 days earlier than what the user entered, as a bit of a failsafe.

The "not" in "Wallet created not before" really throws me off. It's a bit like those "Don't uncheck this checkmark if you don't not want to unsubscribe".

How about simplifying to just month and year and get rid of the date? It's a bit less precise, but easier to remember.

**Comment by Realchacal** [comment]
@GBKS These are some unfathomable excellent ideas! Agreed on all and updated!

**Comment by dEBRUYNE-1** [comment]
>How about simplifying to just month and year and get rid of the date? It's a bit less precise, but easier to remember.

We can default to the height of the first day of the month, that should cover all cases and should not visibly affect user experience (scanning ~20k additional blocks more at most takes a few minutes).

**Comment by Realchacal** [comment]
> ...that should cover all cases
Yes, but potential time zones issues if you've created on the 1st of a month. So maybe generously add 24 hours extra?^^

**Comment by Realchacal** [comment]
Alternative proposal added. Looking for feedback.

**Comment by selsta** [comment]
Why is it marked as optional?

**Comment by Realchacal** [comment]
> Why is it marked as optional?

If the user does not remember when he created the wallet, he can leave this field blank. Then the whole blockchain will have to be scanned.

**Comment by selsta** [comment]
What do you think about moving “Choose your hardware device” above the radio buttons? I would make more sense it is mandatory, changing the radio button selection isn’t.

**Comment by Realchacal** [comment]
@selsta You are absolutely right that would make a lot of sense!! It also would look better visually, because the text boxes are together than!

**Comment by Realchacal** [comment]
![image](https://user-images.githubusercontent.com/46682965/62580490-895a7700-b8a6-11e9-9332-5c3908bd75f8.png)
Maybe we can also change the marked text to "Save wallet to". Because "Wallet location" may sound to a noob like "Specify, where the wallet currently **is**"

**Comment by Realchacal** [comment]
With @selsta's latest idea and some more changes this could give us:
![image](https://user-images.githubusercontent.com/46682965/62603937-3bba2a80-b8f7-11e9-83a5-12839e1ef2ef.png)
Rationale/Explanations:
1: "Name wallet" could be misinterpreted as "Specify the name of your old wallet that you want to restore." Or "Specify the name of your hardware wallet device".
2: "Select" instead of "choose" is more correct here. Also, the label now is above the text box (for consistent design).
3: Nice long path field :) And: Old label "Wallet location" could be misinterpreted as "Specify, where your wallet currently is located."
4: Reduced words. Use of "hardware device" instead of "hardware wallet". Everything in one line.
5: Less complicated then `<major>:<minor` but still 100% correct. And design consistent to text field above.

**Comment by rating89us** [comment]
The "select your hardware wallet" field could be removed if Monero GUI auto-detected whether a Trezor or a Ledger is plugged in.

**Comment by rating89us** [comment]
"First transaction" field should be hidden by default, and should only be displayed if the box "My hardware wallet already holds Monero" is checked.

**Comment by Realchacal** [comment]
> "First transaction" field should be hidden by default, and should only be displayed if the box "My hardware wallet already holds Monero" is checked.

100% agreed! Nice!

**Comment by rating89us** [comment]
I'm against changing "Wallet creation date" to "First transaction", even though first transaction is a more precise term (it's what the wallet really detects when scanning the blockchain):
- Wallet creation will always be on the same date of first transaction or before. Yes, some users may do their first transaction 6-12 months before creating the wallet, but I guess they're the minority.
- The wallet asks users to write down date and block height when they are creating the wallet, and not when the first transaction is done.
- Many users already have paper backups with "wallet creation date" written down. Changing terms would bring confusion.

**Comment by Realchacal** [comment]
> I'm against changing "Wallet creation date" to "First transaction", even though first transaction is a more precise term (it's what the wallet really detects when scanning the blockchain):
> 
> * Wallet creation will always be on the same date of first transaction or before.

Not really. Just think if you are restoring your wallet your 2nd, 3rd time...  Users will have no clue which date to enter. And they have no chance to figure it out by thinking, because "wallet creation date" is just wrong. Or think of compatibility to other software: In the future seeds hopefully will (MUST!) be widely compatible between different wallets. The only thing that will work if everybody does it the right way, which is "First transaction date" - not "Wallet creation date".

> * The wallet asks users to write down date and block height when they are creating the wallet, and not when the first transaction is done.

So if there's a mistake in 2 places this is like saying: "let's not correct the mistake in one place because then it would be inconsistent to the other place"

> * Many users already have paper backups with "wallet creation date" written down. Changing terms would bring confusion.

So we'll be carrying mistakes forever around with us in order to not confuse early adopters?

**Comment by GBKS** [comment]
I'd move this to "Advanced" and call it "Date of first transaction (optional)". To avoid confusion, we can some short explanatory text, either below the label or accessible via a "?" or "Help". We can 

Most of these options cannot be clearly explained to newcomers with 1-3 word labels, so we need a mechanism to get more details inline.

Does "My wallet already holds Monero" make sense if you have to choose to create a new one or import an existing one earlier? New wallets can't hold any yet, and I'd assume the default for existing ones would be that they already have a balance.

**Comment by Realchacal** [comment]
> I'd move this to "Advanced" and call it "Date of first transaction (optional)". To avoid confusion, we can some short explanatory text, either below the label or accessible via a "?" or "Help". We can

100% agreed

> Most of these options cannot be clearly explained to newcomers with 1-3 word labels, so we need a mechanism to get more details inline.

Yes.

> Does "My wallet already holds Monero" make sense if you have to choose to create a new one or import an existing one earlier? New wallets can't hold any yet, and I'd assume the default for existing ones would be that they already have a balance.

Very right. That's why this proposal does not talk about "Create NEW wallet" but instead it says "Create wallet", which is in accordance with #2324. By the way: The term wallet is wrong in general. "Keychain" would be correct. A real wallet holds funds. If you lose it, your funds are gone. You can't copy it. A software wallet doesn't work like this. If you lose it, your funds are still there. You can copy it. That's why it should be called a keychain. Like a real physical keychain, you can copy it. If you lose it, your funds are still there. You just have lost access. But if you have a copy or have the data to create new keys, you still can get your funds.

**Comment by GBKS** [comment]
Yeah, wallet is just a nice metaphor people can make sense of (like a trash can for deleting files). It's always been weird to "download a wallet" and then "create a wallet", with wallet referring to totally different things. I think wallet should be the software you download and set up (or configure), and the keys are what you use to configure it. And either you have keys, or you generate them. But there are lots of opinions on these things...

**Comment by rating89us** [comment]
"_Save wallet to_" should be in Advanced options in this page. When using a hardware wallet, the wallet file is not so important, because the user must have the device, and, if lost, the wallet file can always be recovered later from the device.
